roles/upgrade/tasks/upgrade-validation.yml

```diff
@@ -2,19 +2,21 @@
 - name: Collect updated facts
   ansible.builtin.setup:
 
-- name: Validate current OS version from rhel 7
-  ansible.builtin.assert:
-    that: ansible_distribution_version == rhel_7_upgrade_dest_version
-    fail_msg: Expected leapp destination OS version {{ rhel_7_upgrade_dest_version }} but OS version is {{ ansible_distribution_version }}
-    success_msg: Current OS version is {{ ansible_distribution_version }}.
+- name: Determine leapp_dest_version from rhel 7
+  ansible.builtin.set_fact:
+    leapp_dest_version: "{{ rhel_7_upgrade_dest_version }}"
   when: ansible_facts.ansible_local.pre_ripu.distribution_major_version == '7'
 
-- name: Validate current OS version from rhel 8
+- name: Determine leapp_dest_version from rhel 8
+  ansible.builtin.set_fact:
+    leapp_dest_version: "{{ rhel_8_upgrade_dest_version }}"
+  when: ansible_facts.ansible_local.pre_ripu.distribution_major_version == '8'
+
+- name: Validate current OS version
   ansible.builtin.assert:
-    that: ansible_distribution_version == rhel_8_upgrade_dest_version
-    fail_msg: Expected leapp destination OS version {{ rhel_8_upgrade_dest_version }} but OS version is {{ ansible_distribution_version }}
+    that: ansible_distribution_version == leapp_dest_version
```

Comment:
do you think we should combine this assert with the other one?

Reply:
I'm not sure I understand the request.  The asserts take half a second to run, and it is clear what they are checking for separately.  Based on that I would say no, we shouldn't combine them.